src/browser/atom-application.coffee

```diff
@@ -517,6 +522,8 @@ class AtomApplication
     new AtomWindow({bootstrapScript, @resourcePath, exitWhenDone, isSpec, specDirectory, devMode})
 
   locationForPathToOpen: (pathToOpen) ->
+    {protocol} = url.parse(pathToOpen)
+    return {pathToOpen} if protocol?
```

Comment:
I think this should be moved below the ` return {pathToOpen} unless pathToOpen` check since `url.parse` blows up with a `null` or `undefined` `pathToOpen`